### PR TITLE
Avro with compression

### PR DESCRIPTION
**Pull request by labianchin**

### CODE REVIEW

File: dbeam/src/main/scala/com/spotify/dbeam/JdbcAvroIO.java [hunk 1]
```diff
@@ -150,8 +150,7 @@ private JdbcAvroWriteOperation(JdbcAvroSink sink,
       super(writeOperation, MimeTypes.BINARY);
       this.jdbcAvroOptions = jdbcAvroOptions;
       this.coder = coder;
-      //this.codec = CodecFactory.fromString(this.jdbcAvroOptions.getCodec());
-      this.codec = CodecFactory.deflateCodec(1);
+      this.codec = CodecFactory.deflateCodec(6);
```
Comment (varjoranta) [inline dbeam/src/main/scala/com/spotify/dbeam/JdbcAvroIO.java]: Could you use static field name, like DEFAULT_XZ_LEVEL instead of magic number?

File: dbeam/src/main/scala/com/spotify/dbeam/JdbcAvroIO.java [hunk 2]
```diff
@@ -197,25 +196,22 @@ private ResultSet executeQuery(String query) throws Exception {
...
       try (ResultSet resultSet = executeQuery(query)) {
         checkArgument(resultSet != null,
                       "JDBC resultSet was not properly created");
         this.writeIterateStartTime = System.currentTimeMillis();
         while (resultSet.next()) {
-          GenericRecord genericRecord =
-              jdbcAvroOptions.getAvroRowMapper().convert(resultSet, coder.getSchema());
-          writeRecord(genericRecord);
+          GenericRecord genericRecord = avroRowMapper.convert(resultSet, schema);
+          this.dataFileWriter.append(genericRecord);
```
Comment (varjoranta) [inline dbeam/src/main/scala/com/spotify/dbeam/JdbcAvroIO.java]: The four changes happening here seem to be refactoring and not effecting the logical behavior, and could be separated into different commit.

Comment (labianchin) [reply dbeam/src/main/scala/com/spotify/dbeam/JdbcAvroIO.java]: It is in a separate commit. Do you mean a separate PR also?

Comment (varjoranta) [reply dbeam/src/main/scala/com/spotify/dbeam/JdbcAvroIO.java]: No, you are right. I was reviewing in hurry. Sorry, all hood here.

**Comment by labianchin** [comment]
Also now extracted the compression level into a constant.

File: dbeam/src/main/scala/com/spotify/dbeam/JdbcAvroIO.java [hunk 3]
```diff
@@ -55,7 +55,9 @@
 import org.slf4j.LoggerFactory;
 
 public class JdbcAvroIO {
+
   private static final String DEFAULT_CODEC = "deflate";
+  public static final CodecFactory DEFAULT_DEFLATE_CODEC = CodecFactory.deflateCodec(6);
```
Comment (varjoranta) [inline dbeam/src/main/scala/com/spotify/dbeam/JdbcAvroIO.java]: Ah, again partial confusion on my last hurried review. I understood this number 6 had some other meaning than a fluffy level. I think now this is better as I understand it now as a opinionated level measure on the compressions. Probably cannot make it clearer without explaining where the number came from :)
But good enough! 👍
